packages/widget/src/components/common/buttons/connect-wallet.ts

```diff
@@ -62,49 +62,63 @@ export class ConnectWalletButton extends BaseComponent {
     return !!this.wallets.evmWallet || !!this.wallets.substrateWallet;
   }
 
-  render(): HTMLTemplateResult {
+  private renderConnectWalletButton(): HTMLTemplateResult | undefined {
+    if (this.wallets.substrateWallet) return;
+
+    return when(
+      this.isWalletConnected(),
+      () =>
+        html` <button
+          role="button"
+          @click=${this.onDisconnectClicked}
+          class="connectWalletButton"
+        >
+          Disconnect
+        </button>`,
+      () =>
+        html` <button
+          @click=${this.onConnectClicked}
+          role="button"
+          class="connectWalletButton"
+        >
+          ${plusIcon} Connect Wallet
+        </button>`
+    );
+  }
+
+  private renderWalletAddress(): TemplateResult | undefined {
     const evmWallet = this.wallets.evmWallet;
-    const substrateWallet = this.wallets.substrateWallet;
-    //TODO: this is wrong we need to enable user to select account
-    const substrateAccount = substrateWallet?.accounts[0];
-    return html` <div class="connectWalletContainer">
-      ${when(
-        !!evmWallet?.address,
-        () =>
-          html`<span class="walletAddress" title=${evmWallet?.address ?? ''}
-            >${greenCircleIcon} ${shortAddress(evmWallet?.address ?? '')}</span
-          >`
-      )}
-      ${when(
-        !!substrateAccount,
+    const activeWalletKey = Object.keys(this.wallets).find(
+      (key) => this.wallets[key as WalletContextKeys] !== undefined
+    ) as WalletContextKeys | undefined;
```

Comment:
you can replace this with:

```suggestion
    const activeWalletKey = (
      Object.keys(this.wallets) as (keyof typeof this.wallets)[]
    ).find((key) => !!this.wallets[key]);
```

and still have type checking in lines 97 and 107 (only it wouldn't be done via enum, but string which **must** be key of WalletContext). this would IMO be a bit more future-proof since devs wouldn't need to update `WalletContextKeys` every time `WalletContext` interface is updated. also, this way we avoid typo bugs which could occur since `WalletContextKeys` is manually 'copied' from the interface.

Reply:
Good option as well 👍